### ISSUE TITLE
Issue 010: GPT regression matrix と e2e smoke test を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#359]
# Issue 010: GPT regression matrix と e2e smoke test を追加する

## 背景

coarse GPT は coarse artifact の main pick を変えるため、単体 DP test だけでは不十分。以下をまとめて守る regression が必要。

- GPT disabled では既存挙動が変わらない。
- GPT enabled では `.coarse.npz` main key が GPT 後になる。
- argmax diagnostic key が存在する。
- physics は config key mapping で GPT main / argmax baseline を読める。
- fine は center_npz_files により physics / coarse direct の両方を読める。

## 目的

coarse GPT 実装全体の regression matrix と smoke tests を追加し、今後 3D GPT や fine-direct workflow を検討するときの安全網にする。

## 非目的

この issue では algorithm を新規実装しない。前段 issue の統合確認とテスト強化が目的。

## 変更対象

```text
packages/seisai-engine/tests/test_fbpick_gpt_regression.py
packages/seisai-engine/tests/test_fbpick_coarse.py
packages/seisai-engine/tests/test_fbpick_physics.py
packages/seisai-engine/tests/test_fbpick_fine.py
packages/seisai-engine/tests/test_fbpick_cli.py
```

既存 test に分散させるより、GPT-specific regression は新規 file にまとめることを推奨する。

### A. coarse argmax baseline

```text
coarse GPT disabled
  -> .coarse.npz required keys only
  -> physics input_pick defaults read coarse_pick_i
  -> fine reads physics_window_center_i
```

### B. coarse GPT => physics => fine

```text
coarse GPT enabled
  -> .coarse.npz main coarse_pick_i is GPT result
  -> .coarse.npz has coarse_pick_i_argmax and coarse_pmax_argmax
  -> physics input_pick.pick_i_key=coarse_pick_i
  -> fine center_npz_files=physics.npz
```

### C. argmax diagnostic => physics => fine

```text
same GPT-enabled .coarse.npz
  -> physics input_pick.pick_i_key=coarse_pick_i_argmax
  -> pick_t_sec_key=null derives time from pick_i
```

### D. coarse GPT => fine candidate

```text
coarse GPT enabled
  -> fine center_npz_files=coarse.npz
  -> window_center.npz_key=coarse_pick_i
  -> valid_mask_npz_key=coarse_center_valid_mask
```

### 1. path tracing deterministic tests

既存 Issue 001-003 の tests を regression matrix に含める。

```bash
pytest packages/seisai-engine/tests/test_fbpick_path_tracing_*.py -q
```

### 2. artifact-level test

小さな synthetic payload で `save_coarse_npz()` / `load_coarse_npz()` を通す。

```python
def test_gpt_coarse_npz_can_feed_physics_key_mapping() -> None:
    ...
```

期待:

- `coarse_pick_i` と `coarse_pick_i_argmax` が異なる payload を作る。
- physics `input_pick` を変えると canonical table の pick が変わる。
- `pick_t_sec_key=None` で time が derive される。

### 3. fine center source test

```python
def test_gpt_coarse_npz_can_feed_fine_center_loader() -> None:
    ...
```

期待:

- `center_npz_files=[coarse.npz]`
- `window_center.npz_key=coarse_pick_i`
- `valid_mask_npz_key=coarse_center_valid_mask`
- synthetic info と alignment validation が通る。

### 4. coarse infer fake model smoke

可能なら fake model / fake dataset で `run_coarse_infer()` smoke を作る。

要件:

- logits は row-wise argmax が一部飛ぶように作る。
- GPT enabled では smooth path を選ぶ。
- output `.coarse.npz` に diagnostics が入る。

既存 dataset mocking が重い場合は、`_trace_anchor_predictions_with_gpt()` と restore helper の integration test で代替してよい。

### 5. CLI config parse smoke

```python
def test_cli_config_parse_coarse_gpt_examples() -> None: ...
def test_cli_config_parse_physics_gpt_key_mapping_examples() -> None: ...
def test_cli_config_parse_fine_direct_gpt_examples() -> None: ...
```

## 受け入れ条件

- 上記 A-D の regression matrix が tests / docs で明示される。
- GPT enabled / disabled の差分が明確にテストされる。
- physics を消さずに GPT main / argmax diagnostic の両方を読めることが確認される。
- fine direct candidate が `.coarse.npz` center key で動くことが確認される。

## 推奨確認コマンド

```bash
pytest packages/seisai-engine/tests/test_fbpick_path_tracing_core.py \
       packages/seisai-engine/tests/test_fbpick_path_tracing_dp2d.py \
       packages/seisai-engine/tests/test_fbpick_path_tracing_segmented.py \
       packages/seisai-engine/tests/test_fbpick_coarse.py \
       packages/seisai-engine/tests/test_fbpick_physics.py \
       packages/seisai-engine/tests/test_fbpick_fine.py \
       packages/seisai-engine/tests/test_fbpick_cli.py -q
```

optional deps がない環境では、既存の test stubs / markers に合わせて skip または monkeypatch する。

## 完了条件

- GPT implementation issue series を通して、coarse GPT が opt-in で使える。
- `log_softmax` score 固定。
- `.coarse.npz` main key は GPT 後 pick。
- diagnostics に argmax baseline が残る。
- physics/fine は config key mapping のみで入力切替できる。
- 将来 3D GPT の API slot は残っている。
